src/hash.cr

```diff
@@ -845,6 +876,14 @@ class Hash(K, V)
     raise "Hash table too big"
   end
 
+  private def dig_deep(hash_keys)
```

Comment:
Does extracting the implementation into a private method instead of putting it in the variant taking the array add anything?